### DISCUSSION TITLE
ModuleNotFoundError: No module named 'numpy', but numpy is installed system wide

### DISCUSSION THREAD

**Discussion by jbain1029** [SimDynamX/SpaceTeamsPRO#47] [2024 Lunar SAR Competition Q&A] [answered]
I have been getting this error whenever any of the Python files try to import numpy as np (line 11 I think.)

ModuleNotFoundError: No module named 'numpy'

I have tried installing numpy with pip, conda, and brew, but it still isn't working. Do I need to install it specifically to STPro?

**Comment by connorjak** [comment, marked as answer]
During `FirstTimeSetup.bat`, SpaceTeamsPRO installs a specific python version (currently 3.11.8) if it isn't already installed, and makes a version-specific virtual environment (venv) for use by Python Systems. Numpy and many other packages should be installed by default. Could you run `SpaceCRAFT/FirstTimeSetup.bat` again and check for any errors that come up in the console output? You can keep the output on-screen by opening the Terminal to the SpaceCRAFT folder then running FirstTimeSetup.bat from there.

**Reply by connorjak** [reply]
By the way, this venv is in `AppData/Local/SimDynamX/STPro/Versions/`(the version you have installed)`/py`

**Reply by jbain1029** [reply]
It's working now after I re-ran FirstTimeSetup.bat. Maybe I closed my computer or something before it had a chance to run entirely. Either way, it works now! Thanks!
